ndcube/utils/wcs.py

```diff
@@ -139,76 +140,120 @@ def _wcs_slicer(wcs, missing_axis, item):
-        # then slice(0,1) added else slice(None, None, None) is appended and
+        # Creating a new tuple of slice where if the axis is dead, i.e. missing,
+        # then slice(0,1) added, else slice(None, None, None) is appended; and,
         # if the check of missing_axis gives that this is the index where it
         # needs to be appended then it gets appended there.
-        for _bool in missing_axis:
+        for i, _bool in enumerate(missing_axis):
             if not _bool:
-                if index is not 1:
+                if index is not 1:  
                     item_checked.append(item)
+                    missing_axis[i] = True
                     index += 1
                 else:
                     item_checked.append(slice(None, None, None))
             else:
                 item_checked.append(slice(0, 1))
-        new_wcs = wcs.slice((item_checked))
+        item_ = (item_checked)  # Un-removed brackets to test...
+
     # item is int then slicing axis.
     elif isinstance(item, int) or isinstance(item, np.int64):
-        # using index to keep track of whether the int(which is converted to
+        # Using index to keep track of whether the int(which is converted to
         # slice(int_value, int_value+1)) is already added or not. It checks
-        # the dead axis i.e missing_axis to check if it is dead than slice(0,1)
+        # the dead axis i.e. missing_axis to check if it is dead than slice(0,1)
         # is appended in it. if the index value has reached 1 then the
         # slice(None, None, None) is added.
         index = 0
         for i, _bool in enumerate(missing_axis):
             if not _bool:
-                if index is not 1:
-                    item_checked.append(slice(item, item+1))
+                if index is not 1:  
+                    item_checked.append(slice(item, item + 1))
                     missing_axis[i] = True
                     index += 1
                 else:
                     item_checked.append(slice(None, None, None))
-            else:
+            elif _bool: 
                 item_checked.append(slice(0, 1))
-        new_wcs = wcs.slice(item_checked)
-    # if it a tuple like [0:2, 0:3, 2] or [0:2, 1:3]
+                # item_checked.append(slice(item, item + 1))
+        item_ = item_checked  # Removed brackets to fix bug in code block...
+    # if it a tuple like (0:2, 0:3, 2) or (0:2, 1:3)
     elif isinstance(item, tuple):
         # this is used to not exceed the range of the item tuple
         # if the check of the missing_axis which is False if not dead
         # is a success than the the item of the tuple is added one by
         # one and if the end of tuple is reached than slice(None, None, None)
         # is appended.
         index = 0
-        for _bool in missing_axis:
+        for i, _bool in enumerate(missing_axis):
             if not _bool:
-                if index is not len(item):
+                if index is not len(item): 
```

Comment:
If having ```int```s instead of ```slice``` objects within a ```tuple``` ```item``` is causing this function to not extract the ```dropped_coords```, a solution could be to convert ```int```s within a ```tuple``` ```item``` to ```slice``` objects of length 1, e.g.
```python
if index is not len(item):
    if isinstance(item[index], (int, np.int64)):
        item_checked.append(slice(item[index], item[index]+1))
    else:
       item_checked.append(item[index])
    index += 1
```

In fact, it looks like a similar conversion is already done in the ```int``` part of the overall ```if``` statement: https://github.com/sunpy/ndcube/pull/151/files#diff-b324a6b383c9ce080bf07dcbacda10ddR170
We probably just missed the same need here.

Reply:
Think you are right... I have implemented the changes you suggested, and now the dropped_coords by slicing are being picked up. However, the missing_axis are not changing from `False` to `True` though, and I have no idea how to correct that behavior despite several trials. So now I am seeing the following...

Case 1: `utils_wcs._wcs_slicer(w2, [False, False, False], (slice(0,3), slice(0,3), 1))`
Output:
```
(WCS Keywords
 
 Number of WCS axes: 3
 CTYPE : 'WAVE'  'HPLT'  'HPLN'  
 CRVAL : 1e-09  0.5  1.0  
 CRPIX : -1.0  2.0  2.0  
 PC1_1 PC1_2 PC1_3  : 1.0  0.0  0.0  
 PC2_1 PC2_2 PC2_3  : 0.0  1.0  0.0  
 PC3_1 PC3_2 PC3_3  : 0.0  0.0  1.0  
 CDELT : 2.0000000000000002e-11  0.5  0.4  
 NAXIS : 1  3  3,
 [False, False, False],
 [('custom:pos.helioprojective.lon', None, array(1.))])
```

Case 2: `utils_wcs._wcs_slicer(w2, [False, False, False], (1, slice(0,3), 1)) `
Output:
```
(WCS Keywords
 
 Number of WCS axes: 3
 CTYPE : 'WAVE'  'HPLT'  'HPLN'  
 CRVAL : 1e-09  0.5  1.0  
 CRPIX : -1.0  2.0  1.0  
 PC1_1 PC1_2 PC1_3  : 1.0  0.0  0.0  
 PC2_1 PC2_2 PC2_3  : 0.0  1.0  0.0  
 PC3_1 PC3_2 PC3_3  : 0.0  0.0  1.0  
 CDELT : 2.0000000000000002e-11  0.5  0.4  
 NAXIS : 1  3  1,
 [False, False, False],
 [('em.wl', None, array(1.04e-09)),
  ('custom:pos.helioprojective.lon', None, array(1.))])
```

Case 3: `utils_wcs._wcs_slicer(w2, [False, False, True], (1, slice(0,2)))`
Output:
```
(WCS Keywords
 
 Number of WCS axes: 3
 CTYPE : 'WAVE'  'HPLT'  'HPLN'  
 CRVAL : 1e-09  0.5  1.0  
 CRPIX : 0.0  2.0  1.0  
 PC1_1 PC1_2 PC1_3  : 1.0  0.0  0.0  
 PC2_1 PC2_2 PC2_3  : 0.0  1.0  0.0  
 PC3_1 PC3_2 PC3_3  : 0.0  0.0  1.0  
 CDELT : 2.0000000000000002e-11  0.5  0.4  
 NAXIS : 1  2  1, [True, False, False], [('em.wl', None, array(1.04e-09))])
```

Do you have any hint(s) for me as to how I should debug? E.g., where and how should I place some `missing_axis[i] = True` statements after checking `item_checked`...

Reply:
And, I have a feeling that the following code snippets might not work in our case anymore as it is:
```
for i, it in enumerate(item_checked):
                if isinstance(it, int):
                    missing_axis[i] = True
```